recipes/full_finetune_distributed.py

```diff
@@ -588,8 +587,10 @@ def recipe_main(cfg: DictConfig) -> None:
             "Distributed finetune recipe should be run via a distributed launcher."
             "If using tune CLI, please specify --nnodes 1 and --nproc_per_node [num_gpus]"
         )
-
-    init_process_group(backend="gloo" if cfg.device == "cpu" else "nccl")
+    if torch.xpu.is_available():
```

Comment:
same comment here

Reply:
Thank you for pointing that out! 

Indeed, there are scenarios where the `device_handle` might not be applicable, for example `backend` needs to be determined based on the device type. 

And I think using `cfg.device` as the condition variable may be a simpler way. I will rewrite the code, thank you.